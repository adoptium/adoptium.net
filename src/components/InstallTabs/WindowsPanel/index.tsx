import React from "react"
import ShellBox from "@/components/ShellBox"
import "../InstallTabs.css"

export const PureWindowsPanel = (): React.ReactNode => {
  // Hardcoded value for latest LTS, should be replaced with actual data fetching in Next.js
  const mostRecentLts = "21"
  return (
    <div>
      <ShellBox>
        <span className="install__text__no-select">
          # Install the latest version
        </span>
        <br />
        <span className="install__text__no-select">$</span>
        <span className="install-text-command">winget install </span>
        EclipseAdoptium.Temurin.{mostRecentLts}.JDK
      </ShellBox>
      <br />
      <ShellBox>
        <span className="install__text__no-select">
          # Install a different version
        </span>
        <br />
        <span className="install__text__no-select">$</span>
        <span className="install-text-command">winget install </span>
        EclipseAdoptium.Temurin.11.JDK
      </ShellBox>
      <br />
      <ShellBox>
        <span className="install__text__no-select">
          # Upgrade the Winget package
        </span>
        <br />
        <span className="install__text__no-select">$</span>
        <span className="install-text-command">winget upgrade </span>
        EclipseAdoptium.Temurin.{mostRecentLts}.JDK
      </ShellBox>
      <br />
      <ShellBox>
        <span className="install__text__no-select">
          # Uninstall the Winget package
        </span>
        <br />
        <span className="install__text__no-select">$</span>
        <span className="install-text-command">winget uninstall </span>
        EclipseAdoptium.Temurin.{mostRecentLts}.JDK
      </ShellBox>
      <a
        className="install__docs-button"
        href="https://docs.microsoft.com/en-us/windows/package-manager/winget/"
      >
        Read documentation
      </a>
    </div>
  )
}

const WindowsPanel = (): React.ReactNode => {
  return <PureWindowsPanel />
}

export default WindowsPanel
