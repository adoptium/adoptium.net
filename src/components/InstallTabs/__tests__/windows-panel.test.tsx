import React from "react"
import { render } from "@testing-library/react"
import { describe, expect, it } from "vitest"
import { PureWindowsPanel as WindowsPanel } from "../WindowsPanel"

describe("Tests for WindowsPanel component", () => {
  it("renders correctly", () => {
    const { container } = render(<WindowsPanel />)
    expect(container).toMatchSnapshot()
  })
})
