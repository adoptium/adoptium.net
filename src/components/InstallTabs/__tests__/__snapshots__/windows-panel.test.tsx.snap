// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Tests for WindowsPanel component > renders correctly 1`] = `
<div>
  <div>
    <pre
      class="shell-box no-highlight"
    >
      <div
        class="shell-box-top"
      />
      <code
        class="shell-box-code"
      >
        <span
          class="install__text__no-select"
        >
          # Install the latest version
        </span>
        <br />
        <span
          class="install__text__no-select"
        >
          $
        </span>
        <span
          class="install-text-command"
        >
          winget install 
        </span>
        EclipseAdoptium.Temurin.
        21
        .JDK
      </code>
    </pre>
    <br />
    <pre
      class="shell-box no-highlight"
    >
      <div
        class="shell-box-top"
      />
      <code
        class="shell-box-code"
      >
        <span
          class="install__text__no-select"
        >
          # Install a different version
        </span>
        <br />
        <span
          class="install__text__no-select"
        >
          $
        </span>
        <span
          class="install-text-command"
        >
          winget install 
        </span>
        EclipseAdoptium.Temurin.11.JDK
      </code>
    </pre>
    <br />
    <pre
      class="shell-box no-highlight"
    >
      <div
        class="shell-box-top"
      />
      <code
        class="shell-box-code"
      >
        <span
          class="install__text__no-select"
        >
          # Upgrade the Winget package
        </span>
        <br />
        <span
          class="install__text__no-select"
        >
          $
        </span>
        <span
          class="install-text-command"
        >
          winget upgrade 
        </span>
        EclipseAdoptium.Temurin.
        21
        .JDK
      </code>
    </pre>
    <br />
    <pre
      class="shell-box no-highlight"
    >
      <div
        class="shell-box-top"
      />
      <code
        class="shell-box-code"
      >
        <span
          class="install__text__no-select"
        >
          # Uninstall the Winget package
        </span>
        <br />
        <span
          class="install__text__no-select"
        >
          $
        </span>
        <span
          class="install-text-command"
        >
          winget uninstall 
        </span>
        EclipseAdoptium.Temurin.
        21
        .JDK
      </code>
    </pre>
    <a
      class="install__docs-button"
      href="https://docs.microsoft.com/en-us/windows/package-manager/winget/"
    >
      Read documentation
    </a>
  </div>
</div>
`;
